hilo.rb

```diff
@@ -1,26 +1,28 @@
+# Devtrain: Ruby Hi-Lo
+
 # predefined number of guesses allowed
 guesses = 6
 ## computer secretly picks number between 1 and 100 inclusive
 number = rand(1...101)
 
 # conditional logic that is run as long as user has remaining guesses left
 while guesses > 0
-    # get user input
-    puts "Guess the number between 1 & 100 (inclusive):"
-    # store user input
-    guess = gets.chomp.to_i
-    # if user guesses correctly, game is won (guess = number)
-    if (guess == number)
-        puts "You win! #{number} was the number!"
-        break
-    # if guess too high, decrement number of guesses left and tell them their guess was too high
-    elsif guess > number
-        guesses -= 1
-        puts "Your guess is too high!"
-    # if guess too low, decrement number of guesses left and tell them their guess was too low
-    elsif guess < number
-        guesses -=1
-        puts "Your guess is too Low!"
+  # get user input
+  puts "Guess the number between 1 & 100 (inclusive):"
+  # store user input
+  guess = gets.chomp.to_i
+  # if user guesses correctly, game is won (guess = number)
+  if (guess == number)
+    puts "You win! #{number} was the number!"
+    break
+  # if guess too high, decrement number of guesses left and tell them their guess was too high
+  elsif guess > number
```

Comment:
You've got some really lovely and clear comments here, but part of the ruby philosophy is writing code that doesn't need comments because it's so obvious what it does. We manage this by naming variables and methods well. For example in this case i'd suggest making a method like this
```
def too_high?(guess)
  ..I'll leave you to figure it out
end
```
so then you can just check
```
elseif too_high?(guess)
```
which is very self explanatory.

Note I've named this method with a `?` on the end, this is just a helpful pattern that shows that this method returns a boolean.